app/assets/javascripts/Components/Modals/submission_file_upload_modal.jsx

```diff
@@ -16,18 +17,69 @@ class SubmissionFileUploadModal extends React.Component {
 
   onSubmit = event => {
     event.preventDefault();
-    this.props.onSubmit(this.state.newFiles, this.state.unzip);
+    if (this.state.newFiles.length == 1) {
```

Comment:
This check isn't necessary. You can always pass `this.state.single_file_name` to the `onSubmit` prop, and let it ignore the value if there's more than one file.